### PR TITLE
SERVICE: Created IOrderService and OrderService

### LINKED ISSUE #82: SERVICE: Create OrderService, IOrderService, OrderForUpdateDTO and OrderForViewDTO

### PR DESCRIPTION
**Pull request by AbbosHaydarov**
Closes #82

### CODE REVIEW

File: BelissimoCloneWPF.Service/DTOs/Orders/OrderForViewDTO.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    public class OrderForViewDTO
+    {
+        public string Location { get; set; }
+        public bool IsPayed { get; set; }
+        public OrderType OrderTypes { get; set; }
+        public PaymentType PaymentTypes { get; set; }
+        public int BasketId { get; set; }
+        public Basket Basket { get; set; }
+    }
+}
```
Comment (MuhammadKasimov) [inline BelissimoCloneWPF.Service/DTOs/Orders/OrderForViewDTO.cs:15]: add id property

File: BelissimoCloneWPF.Service/DTOs/Orders/OrderForUpdateDTO.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public class OrderForUpdateDTO
+    {
+        public string Location { get; set; }
+        public bool IsPayed { get; set; }
+        public OrderType OrderTypes { get; set; }
+        public PaymentType PaymentTypes { get; set; }
+        public int BasketId { get; set; }
+    }
+}
```
Comment (MuhammadKasimov) [inline BelissimoCloneWPF.Service/DTOs/Orders/OrderForUpdateDTO.cs:19]: Use Creation instead
remove this class

File: BelissimoCloneWPF.Service/Services/Orders/OrderService.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        public OrderService(IGenericRepository<Order> orderRepository, IMapper mapper)
+        {
+            this.orderRepository = orderRepository;
+            this.mapper = mapper;
+        }
+
+        public async ValueTask<OrderForViewDTO> CreateAsync(OrderForCreationDTO orderForCreationDTO)
+        {
+            var order = await orderRepository.CreateAsync(mapper.Map<Order>(orderForCreationDTO));
+            await orderRepository.SaveChangesAsync();
```
Comment (MuhammadKasimov) [inline BelissimoCloneWPF.Service/Services/Orders/OrderService.cs:30]: chack for basket existance

File: BelissimoCloneWPF.Service/Services/Orders/OrderService.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                throw new BelissimoCloneWPFException(404, "Order not found");
+
+            return mapper.Map<OrderForViewDTO>(order);
+        }
+
+        public async ValueTask<OrderForViewDTO> UpdateAsync(int id, OrderForUpdateDTO orderForUpdateDTO)
+        {
+            var existOrder = await orderRepository.GetAsync(
+               o => o.Id == id);
+
```
Comment (MuhammadKasimov) [inline BelissimoCloneWPF.Service/Services/Orders/OrderService.cs:56]: the same with basket

**Review by MuhammadKasimov — CHANGES_REQUESTED** [review]

File: BelissimoCloneWPF.Service/DTOs/Orders/OrderForViewDTO.cs (hunk 1 above)
Comment (AbbosHaydarov) [reply BelissimoCloneWPF.Service/DTOs/Orders/OrderForViewDTO.cs:15]: Added Id Property

File: BelissimoCloneWPF.Service/DTOs/Orders/OrderForUpdateDTO.cs (hunk 2 above)
Comment (AbbosHaydarov) [reply BelissimoCloneWPF.Service/DTOs/Orders/OrderForUpdateDTO.cs:19]: Removed OrderForUpdateDTO and used OrderForCreationDTO

File: BelissimoCloneWPF.Service/Services/Orders/OrderService.cs (hunk 3 above)
Comment (AbbosHaydarov) [reply BelissimoCloneWPF.Service/Services/Orders/OrderService.cs:30]: Wrote checking BasketId

File: BelissimoCloneWPF.Service/DTOs/Orders/OrderForViewDTO.cs [hunk 5]
```diff
@@ -5,6 +5,7 @@ namespace BelissimoCloneWPF.Service.DTOs.Orders
 {
     public class OrderForViewDTO
     {
+        public int OrderId { get; set; }
```
Comment (MuhammadKasimov) [inline BelissimoCloneWPF.Service/DTOs/Orders/OrderForViewDTO.cs:8]: name it id

**Review by MuhammadKasimov — CHANGES_REQUESTED** [review]
have you removed OrderForUpdateDTO class

File: BelissimoCloneWPF.Service/Services/Orders/OrderService.cs (hunk 4 above)
Comment (AbbosHaydarov) [reply BelissimoCloneWPF.Service/Services/Orders/OrderService.cs:61]: Corrected mistake

File: BelissimoCloneWPF.Service/DTOs/Orders/OrderForViewDTO.cs (hunk 5 above)
Comment (AbbosHaydarov) [reply BelissimoCloneWPF.Service/DTOs/Orders/OrderForViewDTO.cs:8]: changed

**Pull request closed without merging**